backend-shared/util/operations/types.go

```diff
@@ -171,26 +171,28 @@ func createOperationInternal(ctx context.Context, waitForOperation bool, dbOpera
 
 }
 
-// cleanupOperation cleans up the database entry and (optionally) the CR, once an operation has concluded.
+// cleanupOperation cleans up the operation CR and (optionally) the database entry, once an operation has concluded.
 func CleanupOperation(ctx context.Context, dbOperation db.Operation, k8sOperation managedgitopsv1alpha1.Operation, operationNamespace string,
```

Comment:
In several unit tests, we are checking if the Operation rows are created. So, I'm assuming the original intention behind [commenting](https://github.com/redhat-appstudio/managed-gitops/pull/307/files#diff-acb15696e94beedbf9da214a7eee8ce27d9734bce6e5a8ea3ab2f2ac0e165a4cL180-L187) the DB entry deletion was to test the creation of Operation rows.

I've refactored the code to make the DB entry deletion optional instead of the other way around. By doing so, the existing unit tests still pass. Let me know if there will be any unwanted outcomes with this logic.